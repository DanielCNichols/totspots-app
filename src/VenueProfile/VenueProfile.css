.state {
  text-transform: uppercase;
}

.venueProfile_header {
	 background-color: #f14038;
	 padding: 5px;
	 margin: 0;
	 color: white;
}

.link > a {
  color: black;
}

.link > a:visited {
  color: black;
}

.city {
  text-transform: capitalize;
}

.type {
  text-transform: capitalize;
}

.average {
  width: 80%;
  margin:  10px auto;
  display: flex;
  justify-content: space-between;
  padding: 0 10px;
  align-items: center;
}

.contact {
	 width: 80%;
  padding: 0 10px 10px 10px;
  text-align: center;
  margin: 0 auto 0px auto;
	border-bottom: 3px solid #064789;
}

 
.venueProfile {
	border-radius: 3px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
	padding-bottom: 5px;
}


.venueProfile p {
  margin: 5px 0;
}

.ratings {
	margin: 15px auto; 
}

.amenities h3 {
  text-align: center;
}

 .amenities {
			padding: 10px;
			width: 80%;
			border-top: 3px solid #064789;
 }

.amenities li {
  padding: 0 10px;
	width: 60%;
  margin: 5px auto;
  text-align: left;
}

.amenities li:before {
	 color: green;
	 font-weight: 800;
  content: 'âœ“';
  margin-right: 5px;
  font-size: 1.3rem;
}

.venue_controls {
  width: 100%;
  margin: 0 auto;
  display: block;
  justify-content: center;
  text-align: center;
}

.venue_controls button {
  display: block;
  min-width: 180px;
  height: 40px;
  margin: 20px auto;
  border-radius: 4px;
  border-color: blue;
  cursor: pointer;
  background-color: transparent;
  font-size: 1.1em;
}

.rating {
  color: gold;
}

.price {
  color: green;
}

.volume {
  color: black;
}

.NA {
  color: black;
}

@media screen and (min-width: 450px) {

	 .venueProfile {
			margin: 0 auto;
			width: 80%;
	 }


  .venue_controls {
    justify-content: center;
  }
}
@media screen and (min-width: 800px) {

  .venue_controls button {
    min-width: 180px;
    font-size: 1em;
  }
}
