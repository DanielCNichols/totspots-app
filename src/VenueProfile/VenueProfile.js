import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import VenuesContext from '../VenuesContext'

export default class VenueProfile extends Component {
  static contextType =  VenuesContext

  render() {
    let venue = this.context.selectedVenue
    return(
      <section className='venueProfile'>
        <header>
          <h2>{venue.venue_name}</h2>
        </header>
      <p>{venue.venue_type}</p>
      <p>{venue.address}</p>
      <span>{venue.city}</span>, <span>{venue.state}</span><span>{venue.zip}</span>
      <p>`phone`</p>
      <p>`url`</p>
      <div className='ratings'>
        <p>OVERALL AVERAGE RATING</p>
        <p>OVERALL AVERAGE PRICE</p>
        <p>OVERALL AVERAGE VOLUME</p>
      </div>
      <Link to={`/addreview/${venue.id}`}><button>ADD A REVIEW</button></Link>
      <button>Add to Favorites</button> 
      </section>
    )
  }
}