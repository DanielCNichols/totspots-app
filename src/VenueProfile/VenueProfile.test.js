
import React from 'react';
import ReactDOM from 'react-dom'
import VenueProfile from './VenueProfile.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders VenueProfile without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <VenueProfile />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders VenueProfile as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <VenueProfile name="VenueProfile"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
