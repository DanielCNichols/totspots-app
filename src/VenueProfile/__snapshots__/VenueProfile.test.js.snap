// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VenueProfile as expected 1`] = `
<section
  className="venueProfile"
>
  <header
    className="venueProfile_header"
  >
    <h2 />
  </header>
  <div
    className="contact"
  >
    <p
      className="type"
    >
      Type: 
    </p>
    <p />
    <span
      className="city"
    />
    ,
     
    <span
      className="state"
    />
    <span>
       
    </span>
    <p />
    <p
      className="link"
    >
      <a
        rel="noopener noreferrer"
        target="_blank"
      />
    </p>
  </div>
  <div
    className="ratings"
  >
    <div
      className="average"
    >
      Overall Rating:
       
      <span
        className="rating"
      >
        <span
          className="NA"
        >
          Not Rated
        </span>
      </span>
    </div>
    <div
      className="average"
    >
      Overall Price:
       
      <span
        className="price"
      >
        <span
          className="NA"
        >
          Not Rated
        </span>
      </span>
    </div>
    <div
      className="average"
    >
      Overall Volume Level:
       
      <span
        className="volume"
      >
        <span
          className="NA"
        >
          Not Rated
        </span>
      </span>
    </div>
  </div>
  <div
    className="amenities"
  >
    <h3>
      Features
    </h3>
  </div>
  <div
    className="venue_controls"
  >
    <button
      onClick={[Function]}
    >
      Add a review
    </button>
    <button
      onClick={[Function]}
    >
      Add to Saved Venues
    </button>
  </div>
</section>
`;
